deploy/AllocatorInit.sol

```diff
@@ -0,0 +1,278 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.8.0;
+
+import { DssInstance } from "dss-test/MCD.sol";
+import { AllocatorSharedInstance, AllocatorNetworkInstance } from "./AllocatorInstances.sol";
+
+interface WardsLike {
+    function wards(address) external view returns (uint256);
+    function rely(address) external;
+    function deny(address) external;
+}
+
+interface IlkRegistryLike {
+    function put(
+        bytes32 _ilk,
+        address _join,
+        address _gem,
+        uint256 _dec,
+        uint256 _class,
+        address _pip,
+        address _xlip,
+        string calldata _name,
+        string calldata _symbol
+    ) external;
+}
+
+interface RolesLike {
+    function setIlkAdmin(bytes32, address) external;
+    function setUserRole(bytes32, address, uint8, bool) external;
+    function setRoleAction(bytes32, uint8, address, bytes4, bool) external;
+}
+
+interface RegistryLike {
+    function file(bytes32, bytes32, address) external;
+}
+
+interface VaultLike {
+    function ilk() external view returns (bytes32);
+    function roles() external view returns (address);
+    function buffer() external view returns (address);
+    function vat() external view returns (address);
+    function nst() external view returns (address);
+    function file(bytes32, address) external;
+    function draw(uint256) external;
+    function wipe(uint256) external;
+}
+
+interface BufferLike {
+    function approve(address, address, uint256) external;
+}
+
+interface SwapperLike {
+    function roles() external view returns (address);
+    function ilk() external view returns (bytes32);
+    function buffer() external view returns (address);
+    function swap(address, address, uint256, uint256, address, bytes calldata) external;
+}
+
+interface DepositorUniV3Like {
+    struct LiquidityParams {
+        address gem0;
+        address gem1;
+        uint24  fee;
+        int24   tickLower;
+        int24   tickUpper;
+        uint128 liquidity;
+        uint256 amt0Desired;
+        uint256 amt1Desired;
+        uint256 amt0Min;
+        uint256 amt1Min;
+    }
+
+    struct CollectParams {
+        address gem0;
+        address gem1;
+        uint24  fee;
+        int24   tickLower;
+        int24   tickUpper;
+    }
+
+    function roles() external view returns (address);
+    function ilk() external view returns (bytes32);
+    function buffer() external view returns (address);
+    function deposit(LiquidityParams memory) external returns (uint128, uint256, uint256);
+    function withdraw(LiquidityParams memory, bool) external returns (uint256, uint256);
+    function collect(CollectParams memory) external returns (uint256, uint256);
```

Comment:
Do we want these ones here due to the tests?

Reply:
Same for the structs

Reply:
Need these for the function selectors used int `setRoleAction`.

Reply:
Ohh right!